### PR TITLE
fix: Slow plotting

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
## Overview

This PR fixes the plotting operation being slow in the new PrimePlotter object. This is solved by caching some of the operations and updating only when required, after performing a meshing operation in the backend.

### CODE REVIEW

**Comment by AlejandroFernandezLuces** [comment]
@rchopade7 @waltersma whenever you have some time can you take a look at these changes?

File: examples/gallery/04_lucid_toy_car.py [hunk 1]
```diff
@@ -119,9 +127,14 @@
     # Closed part with no hole
     scope = prime.ScopeDefinition(model, part_expression=closed_part.name)
     pl = PrimePlotter()
+    start_time = time.time()
     pl.plot(model, scope)
+    end_time = time.time()
     pl.show()
 
+
+print("Total time for wrapping: ", total_time)
```
Comment (waltersma) [inline examples/gallery/04_lucid_toy_car.py:136]:
```suggestion
print("Total time for plotting: ", total_time)
```

**Comment by mcMunich** [comment]
I am using the 0.6.2 version of pyprime and it takes a really really long time to generate plots.

**Comment by AlejandroFernandezLuces** [comment]
Sorry to hear that @mcMunich , that version was released before this fix was put in place..

@waltersma @rchopade7 do you think it's worth making a patch release with this commit to mitigate the slow plotting problem in 0.6 version?

**Pull request merged**